### ISSUE TITLE
Problem with newest update

### ISSUE THREAD

**Issue by itge13** [ssborbis/ContextSearch-web-ext#45]
![2018-05-29_100813](https://user-images.githubusercontent.com/39725549/40646102-923fd934-6328-11e8-8c42-2ac10097a450.png)
the selected text becomes grey with the newest update, this is very annoying for short copy text with ctrl-c which doesnt work now anymore.

**Comment by ssborbis** [comment]
Ah, first complication from grabbing focus to the search bar. I'll work something out. In the meantime, you might add the Copy to Clipboard tool to your quick menu. Clicking that will copy the text even without the focus.

**Comment by itge13** [comment]
I don't use ctrl-c directly, I programmed my middle mouse button to do so. But it doesn't work without focus. 

Anyway why do we have a text field now? Is a bit redundant imho. An option for deactivating would be nice.

**Comment by hg42** [comment]
I think it is inconsistent to forward keys to the parent, anyways. Other kinds of menus don't work that way.

**Comment by tzeumer** [comment]
Oh, I thought it was a new bug and nearly wrote in the other ticket. Since it is a feature I reformulate:

The "just start typing" does only work when pressing tab once, if the searchbar is at the top (which is the intention now). But if it is at the bottom the focus is set as before. So at the bottom it would break again for this pretty special case.

I liked it better before and think @hg42 has got a point that "capturing" ctrl+c would be a better solution (and indeed probably a useful feature, too). I wonder what copying method (like the middle mouse button in this case) would not send/do a ctrl+c anyway. If it is really done some other way, hiding, tabbing or an extra option are solutions. But maybe the capture method could get a try before - if it weren't too complex?

**Comment by ssborbis** [comment]
Nice catch. Removed focus for search bar on the bottom.

I'm still working out the focus logic. Someone might be doing any number of actions after selecting text in a textbox and not want the focus moved. Pressing tab seemed like a reasonable compromise, as it mimics normal FORM behavior of tab moving to the next input. I did change code to select all text in the searchbar when grabbing focus with tab vs moving the cursor to the end of the text.

**Comment by tzeumer** [comment]
> Someone might be doing any number of actions after selecting text in a textbox and not want the focus moved.
Yes. I think it was a good compromise in the previous version to not move the focus to the searchbar, when selecting text in a "editable" field (input/textarea). 

> Pressing tab seemed like a reasonable compromise, as it mimics normal FORM behavior of tab moving to the next input.
Hmm, the normal behaviour would be to go to the next (page's) form field or whatever tabindex is set. So I'd think in form fields it's ok to require a click to the search bar.

Maybe it helps looking at it per possible access method
* Keyboard: Since the menu does not pop up immediately, setting the focus as before should very likely never be a problem (but for keyboard users it might be a plus); the menu can hardly interfere with any workflow by chance
* Mouse: I think it's the same with any mouse method, because again the menu is only opened if actually wanted. 
* Auto: This method is somewhat likely to have drawbacks, but who choses it, wants a really quick way for searching - and the extension is all about searching. So I think it's ok to focus on this.

Propably everybody would be served if the focus behavior were like before but with just an option to not "steal" the focus. Maybe the search bar settings could get there own "-- Searchbar --" section (moving from Look & Layout) and then the option would be "Autofocus searchbar: 'Immediately start modifying your search terms. Focus is always disabled in form fields. Turning off this option, always keeps the focus on the selected text'"? :)

**Comment by ssborbis** [comment]
Options are probably the best solution. Since it was causing problems with some users, I pushed the update without the focus, which I'll probably keep as the default and provide options like you suggest. I'm trying to balance keeping things simple for novice users and providing complete customization for advanced users. The backend is the easy part. Streamlining the UI is the pain.

**Comment by itge13** [comment]
@ssborbis  everything works perfect again. Thanks for your great and fast effort and this awesome addon!

**Comment by ssborbis** [comment]
Good deal
